gmsh_utils/gmsh_IO.py

```diff
@@ -296,22 +377,23 @@ def generate_gmsh_mesh(self, point_coordinates: Union[List[List[float]], npt.NDA
             None
         """
 
-        #todo add check for clockwise or anticlockwise
-        point_pairs = self.generate_point_pairs(point_coordinates)
+        # todo add check for clockwise or anticlockwise
+        self.generate_point_pairs(point_coordinates)
 
         gmsh.initialize()
         gmsh.model.add(mesh_name)
 
-        if dims == 3:
-            self.make_geometry_3D(point_coordinates, point_pairs, mesh_size, extrusion_length, name_label)
+        for layer in range(len(point_coordinates)):
+            if dims == 3:
+                self.make_geometry_3d(point_coordinates[layer], mesh_size, name_label[layer], extrusion_length)
 
-        elif dims == 2:
-            self.make_geometry_2D(point_coordinates, point_pairs, mesh_size, name_label)
+            elif dims == 2:
+                self.make_geometry_2d(point_coordinates[layer], mesh_size, name_label[layer])
 
-        gmsh.model.geo.synchronize()
+        self.remove_duplicates()
+        gmsh.model.occ.synchronize()
         gmsh.model.mesh.generate(dims)
```

Comment:
I think it is cleaner, if this function is split in two. I.e. Make geometry and generate_mesh. In this way, "generate_mesh" can be reused in case i make the geometry by reading a .geo file